src/main/java/seedu/address/model/Timekeeper.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javafx.collections.ObservableList;
+
+import seedu.address.model.expense.Event;
+import seedu.address.model.expense.Reminder;
+
+/**
+ * Compares system time with upcoming Events.
+ */
+public class Timekeeper {
+    public static final LocalDate SYSTEM_DATE = LocalDate.now();
```

Comment:
Would be good for system_date to not be static in the future, but for now seems to be sufficient